extensions/Worker.Extensions.Http.AspNetCore/src/FunctionsMiddleware/FunctionsHttpProxyingMiddleware.cs

```diff
@@ -47,40 +51,67 @@ public async Task Invoke(FunctionContext context, FunctionExecutionDelegate next
 
             await next(context);
 
-            var invocationResult = context.GetInvocationResult();
-
-            if (invocationResult?.Value is IActionResult actionResult)
+            var responseHandled = await TryHandleHttpResult(context.GetInvocationResult().Value, context, httpContext, true)
+                || await TryHandleOutputBindingsHttpResult(context, httpContext);
+            
+            if (!responseHandled)
             {
-                ActionContext actionContext = new ActionContext(httpContext, httpContext.GetRouteData(), new ActionDescriptor());
-
-                await actionResult.ExecuteResultAsync(actionContext);
+                var logger = context.InstanceServices.GetRequiredService<ExtensionTrace>();
```

Comment:
Do we need to query this service every time or should we consider caching it in the middleware class (which is singleton) level? 

```csharp
private ExtensionTrace? _extensionTracelogger = null;

public async Task Invoke(FunctionContext context, FunctionExecutionDelegate next)
{
    ....
    ....
    _extensionTracelogger = _extensionTracelogger ?? context.InstanceServices.GetRequiredService<ExtensionTrace>();
    _extensionTracelogger.NoHttpResponseReturned(context.FunctionDefinition.Name, context.InvocationId);
}

```

Reply:
This is a good suggestion. We don't **need** to retrieve the service every time, however, this happens in a failure path that is not as likely to occur and wanted to avoid the additional boilerplate required for the marginal difference given the scenario and the fragility that having that uninitialized class level variable introduce here.

This is also part of the reason why this isn't resolved as a constructor parameter by default (i.e., in most cases, it will never be resolved and there would be an additional hit), which would be the preferred approach if we want to have a class level variable.